Documentation/en-us/DataDrivenExamples.md

```diff
@@ -0,0 +1,190 @@
+# Iterate example specifications over a set of data
+
+Sometimes you will find yourself writing unit tests that repeats almost exactly
+the same specification code over and over again. Typically this happens when you
+have a function, class or struct you want to test for different combinations of
+input data and output data.
+
+Instead of repeating almost the exact same code over and over again, you can define
+the data as a set and iterate your test specification.
+
+## Tests for a list of input and outputs
+
+The following is an basic example to illustrate the concept, where we want to test
+the function `double` for a list of input/output combinations:
+
+The source code defining the `double`-function
+```swift
+/* The function we want to test*/
+func double(_ value : Int) -> Int { return 2*value }
+```
+
+The test code defining the data driven specification:
+```swift
+import Quick
+import Nimble
+
+class DoubleFunctionTests: QuickSpec {
+    override func spec() {
+        describe("the double function") {
+            let dataSet : [(input: Int, output: Int)] = [
+                (input: 0, output: 0),
+                (input: 1, output: 2),
+                (input: 2, output: 4)
+            ]
+            dataSet.forEach { (input: Int, output: Int) in
```

Comment:
The types for `input` and `output` will be inferred by based on the content of `dataSet`, so this can be:

```swift
dataSet.forEach { (input, output) in
```